### ISSUE TITLE
 "Max attribute characters per line" is not respected even when "Attribute Tolerance"=5 and "Max attributes per line"=8

### ISSUE THREAD

**Issue by philipag** [Xavalon/XamlStyler#450]
**Describe the bug**
The MACPL setting says that it will work with AT=5 and MAPL=8, but it has no effect.  If I set MACPL=20, then the attributes do not wrap regardless of line length.  Is MACPL  fundamentally broken or am I doing something wrong?

**To Reproduce**
see above

**Expected behavior**
If I have 5 properties which make a line of 100 characters, I expect them to be split into separate lines such that each line contains either only 1 property or <= 20 characters.

**Screenshots**
![image](https://github.com/Xavalon/XamlStyler/assets/6193288/8c3011f2-f35c-4623-a99b-4f4ee0931c58)
![image](https://github.com/Xavalon/XamlStyler/assets/6193288/83ff863b-0b74-4a9c-832d-2e9cacdcf2ed)

**External Configuration**
NA

**Version Info (please complete the following information):**
Version 17.9.0 Preview 1.0, latest XAML Styler

**Comment by grochocki** [comment]
In this case, AT will take precedence. Until you cross AT threshold, MAPL and MACPL will have no effect.

Notes on [MAPL](https://github.com/Xavalon/XamlStyler/wiki/Attribute-Formatting#max-attributes-per-line):
> Defines the maximum number of attributes an element can have on each line after the start tag **if the number of attributes exceeds the [Attribute Tolerance](https://github.com/Xavalon/XamlStyler/wiki/Attribute-Formatting#attribute-tolerance)**.

You can test this by creating an element with _more than_ 5 attributes:

![image](https://github.com/Xavalon/XamlStyler/assets/1414423/a0f8487c-f3d8-49f8-beec-a2bccbb624e1)
 
We do not support a hard column limit, nor do we believe it makes much sense to do so with a markup language like XAML given the many other supported ways of breaking up attributes across lines (#12).

**Comment by philipag** [comment]
@grochocki In my particular case with the 4 attributes that I have ( I cannot add a 5th attribute just to cause wrapping), how can I cause formatting to wrap while keeping up to 5 short attributes (on another element) on the same line?

I tried setting AT=2 and MAPL=8 but that seems to always wrap after 2 attributes no matter what...

**Comment by ylyuanlei0110** [comment]
> In this case, AT will take precedence. Until you cross AT threshold, MAPL and MACPL will have no effect.
> 
> Notes on [MAPL](https://github.com/Xavalon/XamlStyler/wiki/Attribute-Formatting#max-attributes-per-line):
> 
> > Defines the maximum number of attributes an element can have on each line after the start tag **if the number of attributes exceeds the [Attribute Tolerance](https://github.com/Xavalon/XamlStyler/wiki/Attribute-Formatting#attribute-tolerance)**.
> 
> You can test this by creating an element with _more than_ 5 attributes:
> 
> ![image](https://private-user-images.githubusercontent.com/1414423/290137528-a0f8487c-f3d8-49f8-beec-a2bccbb624e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NTczMzcsIm5iZiI6MTcwMzY1NzAzNywicGF0aCI6Ii8xNDE0NDIzLzI5MDEzNzUyOC1hMGY4NDg3Yy1mM2Q4LTQ5ZjgtYmVlYy1hMmJjY2JiNjI0ZTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjdUMDYwMzU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWYyOTA2ZWE2YWUwMTAzMjU1NDg0MWE5YTY2Yjk3NDMwNTMwNjUwY2E2OTcwMzk1N2I3MDdiMTU0NWE3ZDFlYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.AF59lB10T2oNa_X5h169ayOHWXbB9-cydKB_rFT5ptw)
> 
> We do not support a hard column limit, nor do we believe it makes much sense to do so with a markup language like XAML given the many other supported ways of breaking up attributes across lines (#12).

![image](https://github.com/Xavalon/XamlStyler/assets/145285173/5b4e0edf-6e16-4b4a-87ff-5284c1574f96)

How can I separate the two attributes in the first line.........

**Comment by philipag** [comment]
@ylyuanlei0110 Thanks for the clear explanation.  Unfortunately this means there is no way to prevent the line from going far past the screen width in my scenario (I don't want to decrease AT since that will mess up other formatting, and I cannot add properties just for the sake of breaking up the super long line).  This results in poorly formatted lines when 1 property is very long due to complex bindings with converters and parameters, but there < AT other properties on the same line.

**Comment by ylyuanlei0110** [comment]
> @ylyuanlei0110 Thanks for the clear explanation. Unfortunately this means there is no way to prevent the line from going far past the screen width in my scenario (I don't want to decrease AT since that will mess up other formatting, and I cannot add properties just for the sake of breaking up the super long line). This results in poorly formatted lines when 1 property is very long due to complex bindings with converters and parameters, but there < AT other properties on the same line.

Yeah, I'm confused for this problem recently. Seems no way to fix it with XamlStyler(⊙︿⊙)
